test/unit/org/apache/cassandra/net/ConnectionUtils.java

```diff
@@ -240,6 +247,27 @@ private void doCheck(FailCheck testAndFailCheck)
         }
     }
 
+    private static void longCheck(Runnable assertion, long timeout, TimeUnit timeUnit)
+    {
+        long start = System.currentTimeMillis();
+        for (;;)
+        {
+            try
+            {
+                assertion.run();
+                return;
+            }
+            catch (AssertionError e)
+            {
+                long elapsedMs = System.currentTimeMillis() - start;
+                if (elapsedMs > timeUnit.toMillis(timeout))
+                    throw e;
+                else
+                    Uninterruptibles.sleepUninterruptibly(5, TimeUnit.MILLISECONDS);
```

Comment:
may want to take a look at the `org.apache.cassandra.utils.Retry` class.  I created it recently since I didn't find common utils for retries.

```
long maxDurationNanos = timeUnit.toMillis(timeout);
long startNanos = System.nanoTime();
Retry.retryWithBackoffBlocking(Integer.MAX_VALUE, () -> assertion.run(), (cause) -> {
  if (!(cause instanceOf AssertionError)
    return false;
  return (System.nanoTime() - startNanos) <= maxDurationNanos;
});
```

the main differences are:
1) retry doesn't block thread, only result (if you ask for blocking)
2) default to exponential backoff

Reply:
Thanks for the suggestion. I took a close look at the `Retry` utility. 
I think 1) scheduling the check in the other thread does not benefit in this case. Because it needs to block for the check result anyway. 
2) the check operation is not resources intensive (read multiple values and compare). Exponential backoff does not help. In fact, the test can check frequently in order to exit quick. 

There is no api in `Retry` allows adding a different backup off time yet.